gcp_variant_transforms/testing/integration/run_tests_common.py

```diff
@@ -0,0 +1,172 @@
+# Copyright 2018 Google Inc.  All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Common functions and classes that are used by integration tests.
+
+It provides common functions and classes for both run_vcf_to_bq_tests
+(integration test script for vcf_to_bq pipeline) and run_preprocessor_tests
+(integration test script for vcf_to_bq preprocessor pipeline).
+"""
+
+from abc import abstractmethod
+from typing import Dict, List  # pylint: disable=unused-import
+import argparse  # pylint: disable=unused-import
+import json
+import os
+import time
+
+from googleapiclient import discovery
+
+DEFAULT_IMAGE_NAME = 'gcr.io/gcp-variant-transforms/gcp-variant-transforms'
+
+
+class TestCaseFailure(Exception):
+  """Exception for failed test cases."""
+  pass
+
+
+class TestCase(object):
```

Comment:
If you changed the inheritance pattern per my other comment, this class then should probably be called TestRunner or something like that.